src/wound.cpp

```diff
@@ -0,0 +1,226 @@
+    optional( jo, was_loaded, "infects_into", infects_into );
+}
+
+wound::wound( const damage_unit &damage )
+{
+    const auto lookup_iter = wound_type::wound_lookup.find( damage.type );
+
+    if( lookup_iter == wound_type::wound_lookup.cend() ) {
+        debugmsg( string_format( "Error: No wound found that has damage type %s",
+                                 io::enum_to_string( damage.type ) ) );
+        // no point looking through empty vectors
+        return;
+    }
+
+    // damage unit lookup here
+    std::vector<wound_id> potential_hurt;
+    for( const wound_id &wnd : lookup_iter->second ) {
+        if( wnd->min_damage <= damage.amount && wnd->max_damage >= damage.amount ) {
+            potential_hurt.push_back( wnd );
+        }
+    }
+    int pick_random = rng( 0, static_cast<int>( potential_hurt.size() ) - 1 );
+    id = potential_hurt[pick_random];
+    intensity_multiplier = 1.0 + static_cast<double>( damage.amount - id->min_damage ) /
+                           static_cast<double>( id->max_damage - id->min_damage );
+}
+
+std::string wound::name() const
+{
+    return id->name.translated();
+}
+
+std::string wound::description() const
+{
+    return id->description.translated();
+}
+
+wound wound::wound_healed() const
+{
+    if( !heals_into() ) {
+        debugmsg( "Error: wound trying to heal into nonexistent wound" );
+    }
+    wound healed_wound;
+    healed_wound.id = *heals_into();
+    healed_wound.intensity_multiplier = intensity_multiplier;
+    healed_wound.contamination = contamination;
+
+    return healed_wound;
+}
+
+wound wound::wound_infected() const
+{
+    if( !infects_into() ) {
+        debugmsg( "Error: wound trying to infect into nonexistent wound" );
+    }
+    wound infected_wound;
+    infected_wound.id = *infects_into();
+    infected_wound.intensity_multiplier = intensity_multiplier;
+    infected_wound.contamination = contamination;
+
+    return infected_wound;
+}
+
+bool wound::process( const time_duration &t, double healing_factor )
+{
+    age += t * healing_factor * ( 1 - infection );
+    // stand-in sentinel value
+    infection += 0.01 * contamination * to_seconds<double>( t );
+
+    return id->heal_time && age >= *id->heal_time || is_infected();
+}
+
+std::optional<wound_id> wound::heals_into() const
+{
+    return id->heals_into;
+}
+
+std::optional<wound_id> wound::infects_into() const
+{
+    return id->infects_into;
+}
+
+double wound::wound_progression() const
+{
+    if( !id->heal_time ) {
+        // this is a permanent wound
+        return 0.0;
+    }
+    return to_turns<double>( age ) / to_turns<double>( *id->heal_time );
+}
+
+double wound::infection_progression() const
+{
+    return infection;
+}
+
+bool wound::is_infected() const
+{
+    return infection_progression() >= 1.0;
+}
+
+bool wound::overlaps( int target ) const
+{
+    if( location + id->size <= 100 ) {
+        return location >= target && location + id->size <= target;
```

Comment:
```suggestion
        return location <= target && location + id->size >= target;
```

Just fixing the `wound::overlaps` method

Reply:
Thanks a bunch! Gonna handle it with the next pass